### PR TITLE
Add FXIOS-4452 [v105] Submit commits to Glean probe-scraper

### PR DESCRIPTION
**Pull request by relud**
to validate and publish metrics, per https://mozilla.github.io/glean/book/user/adding-glean-to-your-project/enable-data-ingestion.html#validate-and-publish-metrics

> After your product has been enabled, you must submit commits to probe scraper to validate and publish metrics. Metrics will only be published from branches defined in probe scraper's repositories.yaml, or the Git default branch if not explicitly configured. This should happen on every CI run to the specified branches. Nightly jobs will then automatically add published metrics to the Glean Dictionary and other data platform tools.

### CODE REVIEW

**Comment by jevans-mozilla** [comment]
@travis79 Is there context on this PR?  Looks straightforward, but would like some clarity

**Comment by travis79** [comment]
> @travis79 Is there context on this PR? Looks straightforward, but would like some clarity

@relud  is working on some improvements to the Probe Scraper data-tool that does a lot of the heavy lifting involved with data collection , ingestion, and storage, etc.  This is a change that allows the app to take advantage of the new way we are wanting to do things.  @relud could probably elaborate further if I have misrepresented anything.  

tl;dr this change will allow firefox-ios to make use of the new version of a backend tool that is necessary for Glean.

**Comment by relud** [comment]
nothing misrepresented, but from the [same PR on another product](https://github.com/mozilla-rally/rally/pull/47#issuecomment-1164708798):

The main benefit here for Glean products is that this CI job will allow people to see if changes to metrics files are invalid before they are merged.

Another benefit this will soon[^1] provide is that if a glean product does merge invalid changes, that doesn't block schema deploys for all glean products.

[^1]: probe scraper currently runs nightly scraping all of git history for every glean product. When that job fails, all schema deploys are blocked, usually until the offending commits are added to an ignore list in probe scraper. Pending a backend migration planned for 2022H2, this CI job will replace nightly probe scraper runs for this product. Until that migration happens, the nightly job will continue cause new metrics to show up in your analysis environment.

**Comment by badboy** [comment]
@isabelrios Can we get this merged?

**Comment by isabelrios** [comment]
@badboy before landing could you please update the PR with latest main? The CI is failing because there has been changes that are needed to compile. Thanks and sorry it has taken so long to look into this PR

**Comment by badboy** [comment]
Ah, no worried. Let me take a look.

**Comment by badboy** [comment]
> Ah, no worried. Let me take a look.

I can't, as this is in @relud's fork. @relud can you rebase it on top of `main`? Then we land this.

**Comment by relud** [comment]
@isabelrios this was rebased, and checks are now passing.

**Review by lmarceau — APPROVED** [review]
